### PR TITLE
add sender sock addr check to DS packet filter operator

### PR DESCRIPTION
**Pull request by birarda**
- added sender sock addr check to DS packet filter operator

### CODE REVIEW

**Comment by birarda** [comment]
#### Test Notes

The technical side of this was tested by @Atlante45 and I locally (thanks to tcprewrite/tcprelay). 

Please test connecting to the Sandbox of this PR from your local machine. Have a user connect to the Sandbox from outside your network and confirm that they connect as expected. If you have a second machine, connect from inside your network from another machine and confirm that you connect as expected.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2710/HighFidelity-Beta-PR11237-133b7e66b763d6712c7f460a1a89b6b412a68fc6.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2710/HighFidelity-Beta-PR11237-133b7e66b763d6712c7f460a1a89b6b412a68fc6.dmg

File: libraries/networking/src/HifiSockAddr.h [hunk 1]
```diff
@@ -55,6 +55,8 @@ class HifiSockAddr : public QObject {
 
     QString toString() const;
 
+    bool hasPrivateAddress() const; // checks if the address behind this sock addr is private per RFC 1918
```
Comment (Atlante45) [inline libraries/networking/src/HifiSockAddr.h]: `isPrivateAddress` feels like a better description, but that's not a blocker.

Comment (birarda) [inline libraries/networking/src/HifiSockAddr.h]: `HifiSockAddr` was probably a poor name for the class, since there is an address and port inside

Comment (Atlante45) [inline libraries/networking/src/HifiSockAddr.h]: Oh I understand, all good then.
Totally threw me off.

**Comment by cain-kilgore** [comment]
Testing.

**Comment by cain-kilgore** [comment]
# Test Passed
Test | Result
--- | ---
Please test connecting to the Sandbox of this PR from your local machine. | Pass
Have a user connect to the Sandbox from outside your network and confirm that they connect as expected. | Pass
If you have a second machine, connect from inside your network from another machine and confirm that you connect as expected. | Pass

Everything seemed to work okay as expected. Using my laptop as the secondary device on a tethered hotspot to simulate outside the network and using the WiFi to connect to the Sandbox. 

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Pull request merged**